packages/api/lambdas/sf-event-sqs-to-db-records.js

```diff
@@ -135,67 +149,200 @@ const writeExecutionViaTransaction = async ({ cumulusMessage, trx }) =>
-  collection,
-  provider,
   trx,
+  collectionCumulusId,
+  providerCumulusId,
+  executionCumulusId,
 }) =>
   trx(tableNames.pdrs)
     .insert({
       name: getMessagePdrName(cumulusMessage),
       status: getMetaStatus(cumulusMessage),
-      collectionCumulusId: collection.cumulusId,
-      providerCumulusId: provider.cumulusId,
+      collectionCumulusId,
+      providerCumulusId,
+      executionCumulusId,
     })
     .returning('cumulusId');
 
 const writePdr = async ({
   cumulusMessage,
-  collection,
-  provider,
+  collectionCumulusId,
+  providerCumulusId,
+  executionCumulusId,
   knex,
   pdrModel = new Pdr(),
 }) => {
   // If there is no PDR in the message, then there's nothing to do here, which is fine
   if (!messageHasPdr(cumulusMessage)) {
     return undefined;
   }
-  if (!isRecordDefined(collection)) {
-    throw new Error(`Collection reference is required for a PDR, got ${collection}`);
+  if (!collectionCumulusId) {
+    throw new Error('Collection reference is required for a PDR');
   }
-  if (!isRecordDefined(provider)) {
-    throw new Error(`Provider reference is required for a PDR, got ${provider}`);
+  if (!providerCumulusId) {
+    throw new Error('Provider reference is required for a PDR');
   }
   return knex.transaction(async (trx) => {
-    const [cumulusId] = await writePdrViaTransaction({ cumulusMessage, collection, provider, trx });
+    const [cumulusId] = await writePdrViaTransaction({
+      cumulusMessage,
+      collectionCumulusId,
+      providerCumulusId,
+      trx,
+      executionCumulusId,
+    });
     await pdrModel.storePdrFromCumulusMessage(cumulusMessage);
     return cumulusId;
   });
 };
 
-const saveGranulesToDb = async (cumulusMessage) => {
-  const granuleModel = new Granule();
+const writeGranuleViaTransaction = async ({
+  cumulusMessage,
+  granule,
+  collectionCumulusId,
+  providerCumulusId,
+  executionCumulusId,
+  pdrCumulusId,
+  trx,
+}) =>
+  trx(tableNames.granules)
+    .insert({
+      granuleId: granule.granuleId,
+      status: getMetaStatus(cumulusMessage) || granule.status,
+      collectionCumulusId,
+      providerCumulusId,
+      executionCumulusId,
+      pdrCumulusId,
+    });
+
+/**
+ * Write a granule to DynamoDB and Postgres
+ *
+ * @param {Object} params
+ * @param {Object} params.granule - An API granule object
+ * @param {Object} params.cumulusMessage - A workflow message
+ * @param {string} params.collectionCumulusId
+ *   Cumulus ID for collection referenced in workflow message, if any
+ * @param {string} params.executionCumulusId
+ *   Cumulus ID for execution referenced in workflow message, if any
+ * @param {Knex} params.knex - Client to interact with Postgres database
+ * @param {string} [params.providerCumulusId]
+ *   Cumulus ID for provider referenced in workflow message, if any
+ * @param {string} [params.pdrCumulusId]
+ *   Cumulus ID for PDR referenced in workflow message, if any
+ * @param {Object} [params.granuleModel]
```

Comment:
Why did you put `[]` around these?
```
[params.providerCumulusId]
[params.pdrCumulusId]
[params.granuleModel]
```

Reply:
Because they're optional parameters: https://jsdoc.app/tags-param.html#optional-parameters-and-default-values